lib/ansible/modules/system/known_hosts.py

```diff
@@ -119,12 +119,15 @@ def enforce_state(module, params):
 
     found,replace_or_add,found_line,key=search_for_host_key(module,host,key,hash_host,path,sshkeygen)
 
+    params['diff'] = compute_diff(path, found_line, replace_or_add, state, key)
```

Comment:
i agree params is not the correct thing to use for return data .. but it would a separate fix for the module